kong/tools/utils.lua

```diff
@@ -270,6 +270,28 @@ do
     end
   end
 
+
+  -- Recursively URL escape and format key and value
+  -- Handles nested arrays and tables
+  local function recursive_encode_args(parent_key, value, raw, no_array_indexes, query)
+    for sub_key, sub_value in pairs(value) do
+      if no_array_indexes then
+        sub_key = parent_key
```

Comment:
Better not to override the loop-provided variable within the loop's body; using a local is fine and preferred.